pkg/api/handlers/compat/images_build.go

```diff
@@ -115,6 +115,7 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		Memory                  int64    `schema:"memory"`
 		NamespaceOptions        string   `schema:"nsoptions"`
 		NoCache                 bool     `schema:"nocache"`
+		OCIHooksDir             string   `schema:"ocihooksdir"`
```

Comment:
Does this relate to a path on the _host_ ?

Reply:
This is path to the hooks directory on the server end.

Reply:
This should not be allowed to be set on the client side.

Reply:
@rhatdan Did you mean make this hidden on `podman-remote` , the feature was actually requested for use-case where hooks need to be triggered on a podman server using API https://github.com/containers/buildah/issues/4068#issuecomment-1226816211

Reply:
Yes I want hooks triggered on the server side, but we should not be using client side settings or allow client to override.  I would hide the option.

Reply:
I would expect the same behavior as for container create endpoint, where oci hook dir is not possible to override from client side. See https://github.com/containers/buildah/issues/4068#issuecomment-1226047477

Reply:
Build hooks added to buildah is different from the regular podman hooks as podman hooks are being driven from a config file, but build hooks are only triggered when `--hooks-dir` flag is specified and not from a central config, I have made the flag hidden in CLI but only exposed in API so you'll still need to set `ocihooksdir` in API call while making the request.

Reply:
@lukasmrtvy Any chance you can verify the patch ? Since I'm still trying to figure out an easy way to test this in CI.

Reply:
> Yes I want hooks triggered on the server side, but we should not be using client side settings or allow client to override. I would hide the option.

A remote client should not be able to instruct the server to read from or write to any location on the server side.  That's also why certain transports (see `man containers-transports`) are blocked for certain endpoints.

Reply:
I definitely do NOT believe that OCIHooksdir should be available via API.  It should only be configured on the server side. Clients should not be able to change that.

Reply:
@rhatdan Would it be fine if we make `hooksdir` for build containers configurable on `containers.conf` and remote users will only get to choose if they want to use hooks or not and original configuration will be always kept at server end.

Reply:
That sounds reasonable to me.  Users can opt-out from using hooks but cannot control the host.

Reply:
I think hooks dir is currently configurable.

I am not crazy about allowing remote users to opt out of hooks.

Reply:
@rhatdan Only hooks for `podman run` is configurable, we recently added hooks for `build` containers and they are not wired to any cental config as of now.

Reply:
Ok lets wire them up also.